src/lib.rs

```diff
@@ -198,22 +217,7 @@ impl fmt::Display for CanSocketOpenError {
     }
 }
 
-impl error::Error for CanSocketOpenError {
-    fn description(&self) -> &str {
-        match *self {
-            CanSocketOpenError::LookupError(_) => "can device not found",
-            CanSocketOpenError::IOError(ref e) => e.description(),
-        }
-    }
-
-    fn cause(&self) -> Option<&error::Error> {
-        match *self {
-            CanSocketOpenError::LookupError(ref e) => Some(e),
-            CanSocketOpenError::IOError(ref e) => Some(e),
-        }
-    }
-}
-
+impl error::Error for CanSocketOpenError {}
```

Comment:
Since I can't comment on the enum itself since it's not in the diff... could it make more sense to move this error to the `src/err.rs` file since that seems to be the intended spot for most, if not all, error declarations?

Reply:
> A few small comments since it seems like code reorganization is somewhat in scope given the changes so far.

Yes, I'm assuming that this will be a major update with lots of refactoring, and that breaking changes are OK as long as there's an overall consensus that we're moving in the right direction.

> could it make more sense to move this error to the src/err.rs 

Yes, I was thinking the same thing when I reviewed this work recently (the branch is from last year). I may have been thinking to just do a little at a time? But since this PR includes error refactoring as a goal, this is probably a good place to make the change.

Also... I typically prefer to export a single error type from a library, because it tends to make things simpler when you layer libraries on top of each other. And these days I just do it with `thiserror`. But I wasn't sure how people would feel about either of those things.

Reply:
Yeah, this is just pedantic/gut reaction nits, but it seems likely the error design might continue to change as you/whoever keeps working on refactoring/modernization.

End of the day, I guess this is more stream of consciousness than actual suggestion. :P